### PR TITLE
Analytics: allow to download all data

### PR DESCRIPTION
**Pull request by stsewd**
We are selling that we store this data in .com,
but by default we only show data from the last 30 days,
and users can't download this data.

We keep showing data from the last 30 days in the graphs,
but users can download all the data they are allowed to.

There is a PR in .com that overrides these classes.

I have also normalized the dates used for search and page views
to be the same in .org (90 days).

Screenshot 2021-09-15 at 16-44-30 Traffic Analytics Read the Docs

### CODE REVIEW

File: readthedocs/projects/views/private.py [hunk 1]
```diff
@@ -1072,42 +1066,40 @@ def _search_analytics_csv_data(self):
...
             )
 
         file_name = '{project_slug}_from_{start}_to_{end}.csv'.format(
             project_slug=project.slug,
-            start=timezone.datetime.strftime(last_3_month, '%Y-%m-%d'),
+            start=timezone.datetime.strftime(days_ago, '%Y-%m-%d'),
             end=timezone.datetime.strftime(now, '%Y-%m-%d'),
         )
-        # remove any spaces in filename.
-        file_name = '-'.join([text for text in file_name.split() if text])
```
Comment (stsewd) [inline readthedocs/projects/views/private.py:1095]: the generated filename never includes spaces.

File: readthedocs/analytics/tasks.py [hunk 2]
```diff
@@ -77,9 +75,9 @@ def analytics_event(
...
 def delete_old_page_counts():
     """
-    Delete page counts older than 30 days.
+    Delete page counts older than 3 months.
 
     This is intended to run from a periodic task daily.
     """
-    thirty_days_ago = timezone.now().date() - timezone.timedelta(days=30)
-    return PageView.objects.filter(date__lt=thirty_days_ago).delete()
+    days_ago = timezone.now().date() - timezone.timedelta(days=30 * 3)
```
Comment (humitos) [inline readthedocs/analytics/tasks.py:82]: I think that all of these numbers of days should be settings. So, if we decide to change them to 6 months, we have to change only one value.

File: readthedocs/projects/views/private.py [hunk 3]
```diff
@@ -1072,42 +1066,40 @@ def _search_analytics_csv_data(self):
...
                     project=project,
-                    created__date__gte=last_3_month,
-                    created__date__lte=now,
+                    created__date__gte=days_ago,
                 )
                 .order_by('-created')
                 .values_list('created', 'query', 'total_results')
             )
 
         file_name = '{project_slug}_from_{start}_to_{end}.csv'.format(
```
Comment (humitos) [inline readthedocs/projects/views/private.py:1087]: May be good to prefix the filename with `search_` so the user knows what this is data is about.

File: readthedocs/projects/views/private.py [hunk 4]
```diff
@@ -1150,6 +1148,48 @@ def get_context_data(self, **kwargs):
...
+            data = (
+                PageView.objects.filter(
+                    project=project,
+                    date__gte=days_ago,
+                )
+                .order_by('-date')
+                .values_list('date', 'version__slug', 'path', 'view_count')
+            )
+
+        file_name = '{project_slug}_from_{start}_to_{end}.csv'.format(
```
Comment (humitos) [inline readthedocs/projects/views/private.py:1172]: Here the same, maybe this is good with a `traffic_` prefix.

**Review by humitos — COMMENTED** [review]
I didn't do a full review yet, but I think it would be good to add prefixes to the filenames so the user know what data they contain. Besides, I think that all the numbers of days should be settings, so we can easily change them if we decide that we want to retain the data for more/less time in the future

File: readthedocs/analytics/tasks.py (hunk 2 above)
Comment (ericholscher) [reply readthedocs/analytics/tasks.py:82]: Also it needs to vary by plan, in some cases as well. But 👍  on a setting generally.

File: readthedocs/projects/utils.py [hunk 5]
```diff
@@ -56,3 +57,15 @@ class Echo:
     def write(self, value):
         """Write the value by returning it, instead of storing in a buffer."""
         return value
+
+
+def get_csv_file(file_name, csv_data):
```
Comment (ericholscher) [inline readthedocs/projects/utils.py:62]: I think we normally use `filename` elsewhere in the code (and in the header). Let's keep it consistent.

File: readthedocs/projects/views/private.py (hunk 3 above)
Comment (ericholscher) [reply readthedocs/projects/views/private.py:1087]:
```suggestion
        file_name = 'readthedocs_search_{project_slug_{start}_{end}.csv'.format(
```

To make it explicit.

**Review by ericholscher — COMMENTED** [review]
Agreed with Manuel on using a setting here, and we also need to ensure that we're not deleting users data who are paying us to keep it longer.

File: readthedocs/analytics/tasks.py (hunk 2 above)
Comment (stsewd) [reply readthedocs/analytics/tasks.py]: The PR on .com overrides these tasks to take the plan into consideration.

**Review by ericholscher — APPROVED** [review]
This is a great improvement, so let's get it shipped.

**Pull request merged**